fdbclient/include/fdbclient/CommitTransaction.h

```diff
@@ -164,19 +166,21 @@ struct MutationRef {
 	}
 
 	MutationRef encryptMetadata(const std::unordered_map<EncryptCipherDomainId, Reference<BlobCipherKey>>& cipherKeys,
-	                            Arena& arena) const {
-		return encrypt(cipherKeys, SYSTEM_KEYSPACE_ENCRYPT_DOMAIN_ID, arena);
+	                            Arena& arena,
+	                            BlobCipherMetrics::UsageType usageType = BlobCipherMetrics::UNKNOWN) const {
```

Comment:
Ideally I wouldn't put a default value to the method and rather `assert` that Metrics value != UNKNOWN? UNKNOWN isn't useful and feels more like a programming error, imo.

Assert will avoid hiding issues and surface it out sooner in the simulation runs itself.

Reply:
Same comment for all other default value selection for `UNKNOWN` for other methods..

Reply:
I agree in general, surface error sooner is better, but there could be callers that we don't care, like the storage cache, and unit tests. Do you think it make sense to have an OTHER/UNKNOWN category, or want to have the trouble to add category for each of them?

Reply:
Good point. 

For unit test we can add XX_TEST tag and for StorageCache, we should support a specific tag given it is a valid usecase? However, allowing default could let logic bug silently be introduced in the code leading to incorrect metrics in some cases, hence, my resistance.